python_modules/libraries/dagster-databricks/dagster_databricks/databricks_step_main.py

```diff
@@ -54,18 +91,46 @@ def main(
             step_run_ref = pickle.load(handle)
         print("Running dagster job")  # noqa pylint: disable=print-call
 
-        all_events: List[EventLogEntry] = []
+        events_filepath = os.path.dirname(step_run_ref_filepath) + "/" + PICKLED_EVENTS_FILE_NAME
 
-        try:
-            with DagsterInstance.ephemeral() as instance:
-                instance.add_event_listener(step_run_ref.run_id, all_events.append)
-                list(run_step_from_ref(step_run_ref, instance))
-        finally:
-            events_filepath = (
-                os.path.dirname(step_run_ref_filepath) + "/" + PICKLED_EVENTS_FILE_NAME
-            )
+        def put_events(events):
             with open(events_filepath, "wb") as handle:
-                pickle.dump(serialize_value(all_events), handle)
+                pickle.dump(serialize_value(events), handle)
+
+        # Set up a thread to handle writing events back to the plan process, so execution doesn't get
```

Comment:
Should we be doing this with the regular event log writes as well? Given that writing events to the log usually involves remote communication.

Reply:
Yeah definitely seems like a good idea, although potentially the queueing logic would go inside the EventLogStorage classes just so that we could do fancier things like batching writes to DBs? Don't have strong opinions though.